pkg/extension/sumologicextension/extension.go

```diff
@@ -63,10 +65,19 @@ func newSumologicExtension(conf *Config, logger *zap.Logger) (*SumologicExtensio
 }
 
 func (se *SumologicExtension) Start(ctx context.Context, host component.Host) error {
-	// TODO: handle already registered collector; retrieve credentials etc.
-	if err := se.register(ctx); err != nil {
-		return err
+	var err error
+	if se.checkCollectorCredentials() {
+		se.logger.Debug("found stored credentials")
+		if err = se.getCollectorCredentials(); err != nil {
```

Comment:
I don't see that `err` is used later so I suggest:
```suggestion
		if err := se.getCollectorCredentials(); err != nil {
```
and removing `var err error`